src/process_utility.c

```diff
@@ -309,6 +311,137 @@ process_reindex(Node *parsetree)
 }
 
 
+static void
+process_altertable_change_owner(Cache *hcache, AlterTableCmd *cmd, Oid relid)
+{
+	Hypertable *ht;
+	RoleSpec   *role;
+
+	ht = hypertable_cache_get_entry(hcache, relid);
+	if (NULL == ht)
+		return;
+
+	Assert(IsA(cmd->newowner, RoleSpec));
+	role = (RoleSpec *) cmd->newowner;
+	OidFunctionCall2(catalog_get_internal_function_id(catalog_get(), DDL_CHANGE_OWNER),
```

Comment:
I don't see how an upcall is easier or cleaner. Why not just call the convenient `ATExecChangeOwner` function in `commands/tablecmds.c`? Recursing to children tables is really simple, something like:

```
	chunks = find_inheritance_children(relid, NoLock);

	foreach(lc, chunks)
	{
		ATExecChangeOwner(lfirst_oid(lc), new_owner, true, ...);
	}
```
Or, am I missing something? If you look in `tablecmds.c`, there are actually similar easy to use functions for most ALTER TABLE stuff.

In my PR, I also created a convenience function `for_each_chunk()` that takes a function and applies it to all chunks.

Now you need a bunch of boiler plate to do upcalls, and you cannot see within the context of this function what is happening. Instead you need to context switch to PLPGSQL and find the function definition there.